src/common/common.js

```diff
@@ -51,6 +50,22 @@ const jsdoc = new RegExp(/@\w+\b/);
  */
 const jsdocText = new RegExp(/@\w+\b.*/g);
 
+const isClassMethod = '(^\\s*(static|async|static async){0,1}\\s*(?!if\\s)\\w*\\s*\\()'; // prevent matching if statement "if (!x) x = {};"`
```

Comment:
Why not use a ? instead of {0,1}?
I'm not following the exclusion logic of the if statement.  Why specifically if, and not while?

Reply:
The logic for the if statement was for the idiom of us wrapping a block to check if something was already defined or not when we were documenting namespaces.

Reply:
As for the use of quantifier we also want to match the case where its just the name so we would need to make it 
```js
(?:static|async|static async|)
```
which I don't think is actually better.

Reply:
I was suggesting using `(static|async|static async)?`

Reply:
As for "idiom of us wrapping a block to check if something was already defined or not when we were documenting namespaces" where do we do this?  I thought we were doing
`CIQ.XYZ = CIQ.XYZ || function (...) {`

Reply:
There are few places remaining, for example:
```
if (!CIQ.UI) CIQ.UI = {};
```

Reply:
When I removed the `(?!if\\s)` from the regexp it didn't seem to make any difference in the resulting d.ts files.

Reply:
If by removing `(?!if\\s)` nothing changes in generated files it is likely that code flow may have changed or we may have added an additional parameter in JS documentation to indicate constructor and this RegExp is no longer executed on constructors in our codebase.   Given that we have our definition files working as a. reliable test suite for changes we can choose to eliminate now this addition if you find it unnecessary / confusing.